### PR TITLE
!!! TASK: Cleanup deprecated code in FusionService

### LINKED ISSUE #4089: 9.0: TASK: FusionService cleanup old API
With https://github.com/neos/neos-development-collection/pull/3839 the FusionService was refactored. The legacy parts should be removed for Neos 9

### PR DESCRIPTION
**Pull request by mhsdesign**
**Upgrade instructions**

With https://github.com/neos/neos-development-collection/pull/3839 the FusionService was refactored and weird/hardly usable functionally removed. The legacy parts will be removed in this pr.

`FusionService:: getMergedFusionObjectTree` was removed in favour of `FusionService:: createFusionConfigurationFromSite`

In case you used any of the setter methods like `setAppendFusionIncludes` to add additional Fusion includes to the configuration, you should instead build an own service like Monocle did, leveraging the `FusionSourceCodeFactory`: https://github.com/sitegeist/Sitegeist.Monocle/blob/b20da738e2999d66b6e2d10c901d7b1d7b540fc0/Classes/Fusion/FusionService.php#L67

Resolves #4089

**Review instructions**

related to https://github.com/neos/neos-development-collection/pull/4495

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Classes/Domain/Service/FusionService.php [hunk 1]
```diff
@@ -109,156 +71,35 @@ public function createFusionConfigurationFromSite(Site $site): FusionConfigurati
...
      *
      * @return Runtime
      */
     public function createRuntime(
-        Node $currentSiteNode,
-        ControllerContext $controllerContext
-    ) {
+        Site $site,
+        ActionRequest $actionRequest
+    ): Runtime {
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Domain/Service/FusionService.php]: note this method signature is not final and will further change with https://github.com/neos/neos-development-collection/issues/4086

**Review by kdambekalns — APPROVED** [review]
👀

**Review by bwaidelich — APPROVED** [review]
+1 by reading

**Review by mficzel — APPROVED** [review]
Fine be 👀 and testing. Thanks for cleaning up.

**Pull request merged**